de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/EditorManager.java

```diff
@@ -278,6 +247,73 @@ public void resourcesAdded(final IProject project) {
               .invokeAndWait(
                   () -> addProjectResources(project), ModalityState.defaultModalityState());
         }
+
+        /**
+         * Sends awareness information to populate the UserEditorState of the participant that
+         * joined the session.
+         *
+         * <p>This is done by first sending the needed state for all locally open editors and then
+         * sending an activity to the the correct editor as the active editor.
```

Comment:
Does this mean the information of the active editor are send twice or is the active editor not part of the editorPool?

Reply:
The information that the editor is open is send twice for the active editor. The viewport and selection is not send a second time.
The reasoning behind this is that the UserEditorStateManager always sets the last editor that was reported as open/active as the active editor.
See https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/editor/remote/UserEditorState.java#L94

This could be avoided by either sorting the list of open editors to always handle the active editor last or by filtering for the active editor and then handle the complete state sending (including viewport and selection) separately.

I didn't think about it much when implementing this. Activating the locally active editor again in the end is still a relic of the previous implementation.
If you would prefer another option, I can change it.

Reply:
Ah, that makes sense. Could you add this info (last editor that was reported as open/active is the active editor) in your javadoc?

Reply:
Tried to make it more clear what happens. I find it kind of hard to describe it concisely. Does the new javadoc convey all the necessary information?

Reply:
With the new javadoc it is possible to understand the main idea why you send the editor open information twice. I think all necessary information are included.